geomstats/geometry/beta_distributions.py

```diff
@@ -115,8 +117,9 @@ def maximum_likelihood_fit(data, loc=0, scale=1):
         -------
         parameter : array-like, shape=[n_samples, 2]
         """
+        data = gs.cast(data, gs.float32)
         data = gs.to_ndarray(
-            gs.where(data == 1., 1 - EPSILON, data), to_ndim=2)
+            gs.where(data == 1., gs.array(1. - EPSILON), data), to_ndim=2)
```

Comment:
Any chance we could factorize the `gs.array` within the `gs.where` backend function? I assume the `np.where` function does not require a `np.array` as its second argument, so we could mimic this flexibility?

Reply:
that's probably a good thing to do!

Reply:
I'll open an issue for this